packages/theme-predictive-search/src/theme-predictive-search.js

```diff
@@ -0,0 +1,98 @@
+import { validateConfig, validateQuery } from "./validate";
+import request, { parseResponseStatus, parseResponse } from "./request";
+import { debounce, objectToQueryParams, Dispatcher, Cache } from "./utilities";
+
+export default function PredictiveSearch(config) {
+  try {
+    validateConfig(config);
+  } catch (error) {
```

Comment:
Is this catch needed? You're just rethrowing the error?

Reply:
Initially I was planning on having more validation around `config`, but it doesn't scale very well as we add more features to the API. I'll simply let the API return an errors if the config is not valide.